nativelib/src/main/scala/scala/scalanative/posix/sys/types.scala

```diff
@@ -0,0 +1,36 @@
+package scala.scalanative
+package posix.sys
+
+import scala.scalanative.native.{CInt, CStruct0, Ptr, extern}
+
+@extern
+object types {
+
+  type pthread_attr_t = Ptr[CStruct0]
+
+  type pthread_barrier_t = Ptr[CStruct0]
+
+  type pthread_barrierattr_t = Ptr[CStruct0]
+
+  type pthread_cond_t = Ptr[CStruct0]
+
+  type pthread_condattr_t = Ptr[CStruct0]
+
+  type pthread_key_t = Ptr[CStruct0]
+
+  type pthread_mutex_t = Ptr[CStruct0]
+
+  type pthread_mutexattr_t = Ptr[CStruct0]
+
+  type pthread_once_t = Ptr[CStruct0]
+
+  type pthread_rwlock_t = Ptr[CStruct0]
+
+  type pthread_rwlockattr_t = Ptr[CStruct0]
+
+  type pthread_spinlock_t = Ptr[CStruct0]
+
+  type pthread_t = Ptr[CStruct0]
```

Comment:
None of these type aliases above look good to me. All of them are non-pointer values. Please look at how they are defined in sys/types.h C header, and redefine them here accordingly.